### PR TITLE
feat(input-time-zone): add input-time-zone component

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #6590 

## Summary

This adds a new component to allow users to select time zone offsets.

![Screenshot 2023-07-28 at 11 38 14 AM](https://github.com/Esri/calcite-design-system/assets/197440/534cb7b9-27b0-4da7-bef5-752e098a63eb)

## `calcite-input-time-zone` API

### Properties

| Property           | Attribute  | Description                                                                                                                                                               | Type                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         | Default     |
| ------------------ | ---------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------- |
| `disabled`         | `disabled` | When true, prevents user interaction. Notes: This prop should use the @Prop decorator and reflect. The `disabled` Sass mixin must be added to the component's stylesheet. | `boolean`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | `false`     |
| `open`               | `open`                | When `true`, displays and positions the component.                                                                                                                                                                                                                                                                                                                            | `boolean`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    | `false`      |
| `overlayPositioning` | `overlay-positioning` | Determines the type of positioning to use for the overlaid content.  Using `"absolute"` will work for most cases. The component will be positioned inside of overflowing parent containers and will affect the container's layout.  `"fixed"` should be used to escape an overflowing parent container, or when the reference element's `position` CSS property is `"fixed"`. | `"absolute" \| "fixed"`                                         
| `scale`            | `scale`    | Specifies the size of the component.                                                                                                                                      | `"l" \| "m" \| "s"`                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | `"m"`       |
| `value`              | `value`               | The component's value, where the value is the time zone offset or the difference, in minutes, between the selected time zone and UTC. If no value is provided, the user's time zone offset will be selected by default.                                                                                                                                                                                                                                         | `string`

### Events

| Event                        | Description | Type                |
| ---------------------------- | ----------- | ------------------- |
| `calciteInputTimeZoneChange` | Fires when the time zone offset value has changed. | `CustomEvent<void>` |

### Methods

#### `setFocus() => Promise<void>`

##### Returns

Type: `Promise<void>`

### CODE REVIEW

**Review by driskull — APPROVED** [review]
👍 💯

**Comment by jcfranco** [comment]
Moving to draft to iron out possible issue related to associated time zone grouping data per release.

File: packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+   */
+  @Prop({ reflect: true }) overlayPositioning: OverlayPositioning = "absolute";
+
+  /** Specifies the size of the component. */
+  @Prop({ reflect: true }) scale: Scale = "m";
+
+  /**
+   * The component's value, where the value is the time zone offset or the difference, in minutes, between the selected time zone and UTC.
+   *
+   * If no value is provided, the user's time zone offset will be selected by default
```
Comment (geospatialem) [inline packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx:114]: nitpick:

Add a period at the end of the line:

> If no value is provided, the user's time zone offset will be selected by default.

**Review by geospatialem — COMMENTED** [review]
One small nitpick on the doc. Looking forward to this, nice work!! ⏳

File: packages/calcite-components/src/components/input-time-zone/input-time-zone.stories.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    options: {
+      timezone: "America/Los_Angeles",
+    },
+  },
+  ...storyFilters(),
+};
+
+export const simple = (): string => html`
+  <calcite-input-time-zone>
+    ${boolean("disabled", false)} scale="${select("scale", ["s", "m", "l"], "m")}"
```
Comment (driskull) [inline packages/calcite-components/src/components/input-time-zone/input-time-zone.stories.ts:20]: I think this needs to go inside the tag. Can you verify its working?

File: packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+  assetsDirs: ["assets"],
+  shadow: {
+    delegatesFocus: true,
+  },
+})
+export class InputTimeZone
+  implements
+    InteractiveComponent,
+    LabelableComponent,
+    LoadableComponent,
```
Comment (driskull) [inline packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx:53]: Does this need to implement `FormComponent`?

File: packages/calcite-components/src/components/input-time-zone/readme.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
...------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------ |
```
Comment (driskull) [inline packages/calcite-components/src/components/input-time-zone/readme.md:8]: Should we add any usage markdown examples? Should that be a requirement for new components/features in a PR?

File: packages/calcite-components/src/components/input-time-zone/input-time-zone.stories.ts (hunk 2 above)
Comment (driskull) [reply packages/calcite-components/src/components/input-time-zone/input-time-zone.stories.ts:20]: Should there be a story where its open by default?

**Review by driskull — APPROVED** [review]
👍

File: packages/calcite-components/src/components/input-time-zone/input-time-zone.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  describe("selects user's matching timezone offset by default", () => {
+    const timeZoneNamesAndOffsets = [
+      { name: "America/Los_Angeles", offset: -420 },
+      { name: "Europe/London", offset: 60 },
+    ];
+
+    timeZoneNamesAndOffsets.forEach(({ name, offset }) => {
+      it(`selects default timezone for "${name}"`, async () => {
+        const page = await newE2EPage();
+        await page.emulateTimezone(name);
```
Comment (eriklharper) [inline packages/calcite-components/src/components/input-time-zone/input-time-zone.e2e.ts:73]: very cool, didn't know this was a thing!

File: packages/calcite-components/src/components/input-time-zone/input-time-zone.e2e.ts [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    disabled("calcite-input-time-zone", { shadowAriaAttributeTargetSelector: "calcite-combobox" });
+  });
+
+  describe("t9n", () => {
+    t9n("calcite-input-time-zone");
+  });
+
+  describe("selects user's matching timezone offset by default", () => {
+    const timeZoneNamesAndOffsets = [
+      { name: "America/Los_Angeles", offset: -420 },
```
Comment (benelan) [inline packages/calcite-components/src/components/input-time-zone/input-time-zone.e2e.ts:66]: nice

File: packages/calcite-components/src/components/combobox/combobox.tsx [hunk 7]
```diff
@@ -870,7 +870,7 @@ export class Combobox
         }
       });
 
-      this.filteredItems = this.getfilteredItems();
+      this.filteredItems = this.getFilteredItems();
```
Comment (benelan) [inline packages/calcite-components/src/components/combobox/combobox.tsx:873]: :pray:

File: packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+  }
+
+  render(): VNode {
+    return (
+      <Host>
+        <calcite-combobox
+          clearDisabled={true}
+          disabled={this.disabled}
+          label={this.messages.chooseTimeZone}
+          onCalciteComboboxBeforeClose={this.onComboboxBeforeClose}
```
Comment (eriklharper) [inline packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx:261]: should we be passing down the component's `lang` property to the combobox here?

File: packages/calcite-components/src/components/input-time-zone/readme.md (hunk 4 above)
Comment (jcfranco) [reply packages/calcite-components/src/components/input-time-zone/readme.md:8]: I added a basic, basic usage example (badum tss 🥁). This should get picked up by the doc updating action, right? 

@benelan In terms of short description, do you have a suggestion or could you point me to an existing example to follow?

File: packages/calcite-components/src/components/input-time-zone/input-time-zone.scss [hunk 9]
```diff
@@ -0,0 +1,3 @@
+:host {
+  display: block;
+}
```
Comment (driskull) [inline packages/calcite-components/src/components/input-time-zone/input-time-zone.scss:3]: I think this should import the `@include base-component();` mixin in order to support `:host([hidden]) {` I wonder why its working in the test now.

File: packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx [hunk 10]
```diff
@@ -0,0 +1,291 @@
...
+    /* wired up by t9n util */
+  }
+
+  /**When `true`, displays and positions the component. */
+  @Prop({ mutable: true, reflect: true }) open = false;
+
+  /**
+   * Determines the type of positioning to use for the overlaid content.
+   *
+   * Using `"absolute"` will work for most cases. The component will be positioned inside of overflowing parent containers and will affect the container's layout.
```
[the remark this answers is not in the document]
Comment (benelan) [reply packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx:117]: I don't think linters will wrap comments which is why I asked. People would have to manually break the lines

**Review by benelan — APPROVED** [review]
Are we not adding the `numbering-system` attribute to this component, or is it going to be a follow up item? Otherwise LGTM

File: packages/calcite-components/src/components/input-time-zone/input-time-zone.scss (hunk 9 above)
Comment (jcfranco) [reply packages/calcite-components/src/components/input-time-zone/input-time-zone.scss:3]: Nice catch. 

I think that test helper needs some looking into.

File: packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (jcfranco) [reply packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx:117]: Ah, gotcha. Linting could warn/force us to keep max length in check.

ESLint's [`max-len`](https://eslint.org/docs/latest/rules/max-len) [seems promising](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbWF4LWxlbjogW1wiZXJyb3JcIiwgeyBcImNvZGVcIjogODAgfV0qL1xuXG4vKipcbiAqIFBva2VtIGlwc3VtIGRvbG9yIHNpdCBhbWV0IENydXN0bGUgVGF1cm9zIE1hZ2NhcmdvIHlvdSdyZSBub3Qgd2VhcmluZyBzaG9ydHNcbiAqIFB1cnVnbHkgR2xpZ2FyLiBKb2h0byBDbGVmZmEgUG9saXdhZyBKZWxsaWNlbnQgQnVybXkgU2VhbGVvIFNtb29jaHVtLiBSYXpvciBcbiAqIExlYWYgRXNwZW9uIFBvb2NoeWVuYSBDcnVzdGxlIEFzaCBLZXRjaHVtIE5pZG9raW5nIExhcnZpdGFyLiBXaW5nIEF0dGFjayBcbiAqIFBpcGx1cCBjb25zZWN0ZXR1ciBhZGlwaXNpY2luZyBlbGl0IFRyZWVja28gRnJvc2xhc3MgU2FsYW1lbmNlIEVzY2FwZSBSb3BlLiBcbiAqIFNvbmljIEJvb20gVHJvcGl1cyBFc2NhcGUgUm9wZSBTbW9vY2h1bSBTY29saXBlZGUgQ2F0ZXJwaWUgUGxhaW4gQmFkZ2UuXG4gKi9cbnZhciBmb28gPSB7XG4gIFwiYmFyXCI6IFwiVGhpcyBpcyBhIGJhci5cIixcbiAgXCJiYXpcIjogeyBcInF1eFwiOiBcIlRoaXMgaXMgYSBxdXhcIiB9LFxuICBcImVhc2llclwiOiBcInRvIHJlYWRcIlxufTtcbiIsIm9wdGlvbnMiOnsiZW52Ijp7fSwicnVsZXMiOnt9LCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e30sImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCJ9fX0=).

**Comment by jcfranco** [comment]
> Are we not adding the numbering-system attribute to this component, or is it going to be a follow up item? Otherwise LGTM

I knew I forgot something else. 🫠 I'll add this in a follow-up PR.

File: packages/calcite-components/src/components/input-time-zone/input-time-zone.scss (hunk 9 above)
Comment (jcfranco) [reply packages/calcite-components/src/components/input-time-zone/input-time-zone.scss:3]: Never mind, the helper is working fine. Tests just need a-fixin'. 🔧

**Pull request merged**